File: ros2_demo_py/ros2_demo_py/listener.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import Int64, String

class Listner(Node):
    def __init__(self):
        super().__init__("ROS2_Listener")

        self.subscriber_ = self.create_subscription(String, "message", self.callback, 10)
        # self.timer_ = self.create_timer(0.5, self.publish_msg)

        self.get_logger().info("Hello From ROS2 Listener!!")

    def callback(self, msg):
        self.get_logger().info("Message: " + str(msg.data))


def main():
    rclpy.init()
    node = Listner()
    rclpy.spin(node)
    rclpy.shutdown()
    

if __name__ == "__main__":
    main()

File: ros2_demo_cpp/src/listener.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class Listener : public rclcpp::Node
{
public:
    Listener(): Node("listener")
    {
        subscriber_ = this->create_subscription<std_msgs::msg::String>("message", 10, std::bind(&Listener::callback, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "Hello From ROS2 Listener!!");
    }

private:
    void callback(const std_msgs::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), msg->data);
    }

    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    
    auto node = std::make_shared<Listener>();
    rclcpp::spin(node);

    rclcpp::shutdown();
    return 0;
}